sdk/core/core/test/cmocka/test_az_span.c

```diff
@@ -759,6 +759,62 @@ static void az_span_copy_empty(void** state)
   assert_true(az_span_is_content_equal(az_span_copy(dst, AZ_SPAN_NULL), dst));
 }
 
+static void az_span_trim(void** state)
+{
+  (void)state;
+  az_span source = _az_span_trim_white_space(AZ_SPAN_FROM_STR("   abc   "));
+  assert_true(az_span_is_content_equal(source, AZ_SPAN_FROM_STR("abc")));
+}
+
+static void az_span_trim_left(void** state)
+{
+  (void)state;
+  az_span source = _az_span_trim_white_space_from_start(AZ_SPAN_FROM_STR("   abc   "));
+  assert_true(az_span_is_content_equal(source, AZ_SPAN_FROM_STR("abc   ")));
+}
+
+static void az_span_trim_right(void** state)
+{
+  (void)state;
+  az_span source = _az_span_trim_white_space_from_end(AZ_SPAN_FROM_STR("   abc   "));
+  assert_true(az_span_is_content_equal(source, AZ_SPAN_FROM_STR("   abc")));
+}
+
+static void az_span_trim_all_white(void** state)
+{
+  (void)state;
+  az_span source = _az_span_trim_white_space(AZ_SPAN_FROM_STR("\t\n\r       "));
+  assert_int_equal(az_span_size(source), 0);
+}
+
+static void az_span_trim_none(void** state)
```

Comment:
Add a test where other characters that might be considered "white space" don't get trimmed. For example NO-BREAK SPACE (U+00A0), LINE SEPARATOR character (U+2028), etc.

Within Unicode, the space category contains a bunch of things:
https://www.fileformat.info/info/unicode/category/Zs/list.htm

https://docs.microsoft.com/en-us/dotnet/api/system.char.iswhitespace?view=netcore-3.1#System_Char_IsWhiteSpace_System_Char_